### PR TITLE
Overhauls logging subsystem

### PR DESCRIPTION
**Pull request by darrylmasson**
Redoes the logging subsystem. Main changes:
- A few more messages at the DEBUG level, mainly concerning the scheduling of commands and their execution on the physical hardware.
- All messages go to disk: DEBUG goes to one file, INFO and higher to another
- WARNING and higher goes to the database
- Logfiles will be written to a centrally-located disk for ease of access (NB - this disk needs to be configured and mounted, also we should write some supporting scripts for managing these)

### CODE REVIEW

**Review by jarongrigat — COMMENTED** [review]
Looks good to me. Found a few typos.

File: Doberman/utils.py [hunk 1]
```diff
@@ -162,56 +164,83 @@ def interrupt(self, *args):
 
 class DobermanLogger(logging.Handler):
     """
-    Custom logging interface for Doberman. Logs to
-    the database (with disk as backup).
+    Custom logger for Doberman. DEBUG goes to disk, INFO and higher also to disk in own files,
```
[the remark this answers is not in the document]
Comment (darrylmasson) [reply Doberman/utils.py:167]: DEBUG gets one file, everything else gets one file. I'll clean up the comment.

File: Doberman/utils.py [hunk 2]
```diff
@@ -162,56 +164,83 @@ def interrupt(self, *args):
...
-        self.db = None
-
-    def __del__(self):
-        self.close()
+        self.today = datetime.date.today()
+        self.open_files(self.today)
+        self.flush_cycle = 0
+
+    def rotate(self, when):
+        for f in cls.output_files.values():
```
[the remark this answers is not in the document]
Comment (darrylmasson) [reply Doberman/utils.py:183]: Yes

File: Doberman/utils.py [hunk 3]
```diff
@@ -162,56 +164,83 @@ def interrupt(self, *args):
...
-            name=record.name,
-            funcname=record.funcName,
-            lineno=record.lineno)
-        if self.db.insert_into_db(self.db_name, self.collection_name, rec):
-            self.backup_logger.emit(record)
-
-
-def logger(name, db, loglevel='DEBUG'):
+        msg_datetime = datetime.datetime.fromtimestamp(record.created)
+        msg_date = datetime.date(mgs_datetime.year, msg_datetime.month, msg_datetime.day)
```
[the remark this answers is not in the document]
Comment (darrylmasson) [reply Doberman/utils.py:210]: Whoops

File: Doberman/utils.py [hunk 4]
```diff
@@ -162,56 +164,83 @@ def interrupt(self, *args):
...
+        m = self.format_message(msg_datetime, record.levelname, record.funcName, record.lineno, record.getMessage())
+        with self.mutex:
+            # we wrap anything hitting files or stdout with a mutex because logging happens from
+            # multiple threads
+            if msg_date != self.today:
+                # it's a brand new day, and the sun is high...
+                self.rotate(msg_today)
```
[the remark this answers is not in the document]
Comment (darrylmasson) [reply Doberman/utils.py:217]: Whoops

**Review by jarongrigat — APPROVED** [review]

**Comment by darrylmasson** [comment]
This is merged into #115 so we don't need this PR

**Pull request closed without merging**